analysis/demonhuntervengeance/src/modules/shadowlands/legendaries/BlindFaith.tsx

```diff
@@ -60,12 +99,71 @@ export default class BlindFaith extends Analyzer {
 
     const nextStacks = event.numberofSoulFragmentsConsumed + this.currentStacks;
     this.currentStacks = nextStacks > 20 ? 20 : nextStacks;
+
+    const prevStacks =
+      nextStacks > 1
+        ? Object.keys(this.dataset)
+            .map((key) => Number.parseInt(key))
+            .reduce((acc, key) => (acc > key ? acc : key), 0)
+        : 0;
+
+    // if numberOfSoulFragmentsConsumed > 1 we're skipping stacks
+    if (prevStacks in this.dataset) {
+      this.dataset[prevStacks].end = event.timestamp;
+    }
+
+    this.dataset[nextStacks] = { start: event.timestamp, end: 0 };
   }
 
-  statistic() {
-    const avg = Math.round(
-      this.stacksGained.reduce((acc, stacks) => acc + stacks, 0) / this.stacksGained.length,
+  goodWindowsPercentage = 0;
```

Comment:
I doubt this is correct? 

Also it's hard to review this if more features keep piling on :(

Reply:
> I doubt this is correct?

what should it be, a getter fn? I've loosely based this on spirit bomb and didnt look around tooooo much for this one tbh

> Also it's hard to review this if more features unrelated to existing PR keeps piling on after every review :(Also it's hard to review this if more features unrelated to existing PR keeps piling on after every review :(

sorry, had a burst of productivity in a sleepless night there, further stuff will def be split up